_posts/2017-10-24-Bun.md

```diff
@@ -0,0 +1,61 @@
+---
+title: Fuzzing for CI Workflows
+layout: post
+author: yomimono
+category: General
+tags: ocaml fuzzing afl crowbar travis ci platform tooling testing
+---
+
+# `ocaml-bun`: Fuzzing for CI
+
+Stephen Dolan recently presented [crowbar](https://github.com/stedolan/crowbar) at the 2017 OCaml Workshop. Crowbar bridges a gap between property-based testing frameworks and instrumentation-based automated testing techniques.  Tests written in Crowbar can be executed by the wildly popular and successful [American Fuzzy Lop](http://lcamtuf.coredump.cx/afl) fuzzer.
+
+American Fuzzy Lop provides a command-line tool for running tests, `afl-fuzz`.  `afl-fuzz` will invoke one test instance to run a given test continuously on different inputs, keeping track of which inputs generated crashes (or, for Crowbar, were counterexamples for a given proposition).  The default settings for `afl-fuzz` don't terminate when crashes are discovered, or when new execution paths haven't been found.  The user is expected to terminate all the `afl-fuzz` instances themselves when they're satisfied that enough work has been done (as indicated, colorfully, in the excellent GUI), and then inspect the files left by `afl-fuzz` during the run.
```

Comment:
very minor: afl-fuzz is repeated at the end and start of first two sentences, consider rephrasing to avoid